Fundamentals/Backoffice/Property-Editors/Built-in-Property-Editors/Block-Editor/build-custom-view-for-blocks.md

```diff
@@ -0,0 +1,133 @@
+
+## Write your own HTML view
+
+Currently you can only pick HTML files for a custom view. These views are powered by AngularJS and you can write any AngularJS logic.
+
+Your HTML can be anything. If you use it as a representation of the content you would also want the full view to be clickable. This would then open the default editor for editing of your content.
+
+The following example displays the property with the alias `headline` together with the `description` inside a button to edit your block.
+
+```html
+<button type="button" ng-click="block.edit()">
+  <h2 ng-bind="block.data.headline"></h2>
+  <p ng-bind="block.data.description"></p>
+</button>
+```
+
+If you would like to display properties of `settings`, you can access these by using `block.settingsData.myPropertyAlias`.
+
+## Add the Areas Container for Block Grid Editor
+
+Blocks of the Block Grid Editor can have Areas. These Blocks requires their Custom View to define where the Area Container should be inserted.
+
+This enables you to wrap and place the Areas as you need.
+
+The Area Container is inserted through a slot. Insert the following snippet where you would like the Area to appear in your Custom View:
+
+```html
+<slot name="areas" part="areas"></slot>
+```
+
+## Adding custom implementation to your View
+
+To achieve this you need to add a custom AngularJS controller to your custom view, using the `ng-controller` attribute:
+
+```html
+<button type="button" ng-controller="customBlockController" ng-click="api.editBlock(block, block.hideContentInOverlay, index, parentForm)" class="btn-reset umb-outline blockelement-labelblock-editor blockelement__draggable-element ng-scope ui-sortable-handle" >
+    <h2 ng-bind="block.data.headline"></h2>
+    <p ng-bind="block.data.description"></p>
+</button>
+```
+
+:::note
+The class declaration is necessary to allow the block to behave like the default blocks, like the "drag-to-sort" feature.
+:::
+
+Create a folder inside the `App_Plugins` folder called 'CustomBlockView'.
+
+Create two files within the CustomBlockView file: `package.manifest` and `customBlock.controller.js`.
+ 
+Add the following JSON to the `package.manifest` file:
+
+```json
+{
+  "javascript": [
+    "~/App_Plugins/CustomBlockView/customBlock.controller.js"    
+  ]  
+}
+```
+
+:::note
+Umbraco will parse all `package.manifest` files and load any resources they reference into the backoffice during startup.
+:::
+
+The second file, `customBlock.controller.js`, will be used to register the 'customBlockController' defined using the `ng-controller` attribute in your custom view.
+
+To register the controller, add the following lines of code:
+
+```javascript
+angular.module("umbraco").controller("customBlockController", function ($scope) {
+// you can do your custom functionality here!
```

Comment:
@nielslyngsoe be more explicit or give some common examples on why you would want to use custom AngularJS code rather than this comment. People like to learn by examples and be inspired on what they can achieve.

Reply:
Ignore this comment as I see there is an example below